### PR TITLE
Fix SegmentCompletionProtocol to ignore unknown json fields

### PR DESCRIPTION
**Pull request by mcvsubbu**
Our compatibility guarantees are around upgrading the controller first
and then the broker and then the server. So, it is possible that the
controller sends a new field in the response (in SegmentCompletionProtocol)
The server should ignore new fields that it does not understand

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/protocols/SegmentCompletionProtocol.java [hunk 1]
```diff
@@ -395,6 +396,7 @@ public SegmentStoppedConsuming(Params params) {
     public Response() {
     }
 
+    @JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/protocols/SegmentCompletionProtocol.java:399]: This line should be added before line 388

Comment (mcvsubbu) [reply pinot-common/src/main/java/org/apache/pinot/common/protocols/SegmentCompletionProtocol.java:399]: good catch, thanks

**Comment by mcvsubbu** [comment]
Issue #5359 

This change must go into 0.4.0

**Comment by mcvsubbu** [comment]
@haibow please make sure this commit gets into 0.4.0 thanks

**Pull request merged**